config-vault/src/main/java/io/scalecube/config/vault/VaultConfigSource.java

```diff
@@ -28,99 +30,99 @@
+    this.secretsPaths = new ArrayList<>(secretsPaths);
   }
 
   @Override
   public Map<String, ConfigProperty> loadConfig() {
     Map<String, ConfigProperty> result = new HashMap<>();
-    for (String path : secretsPath) {
+    for (String path : secretsPaths) {
       try {
         LogicalResponse response = vault.invoke(vault -> vault.logical().read(path));
-        final Map<String, LoadedConfigProperty> pathProps = response.getData().entrySet().stream()
-            .map(LoadedConfigProperty::withNameAndValue)
-            .map(LoadedConfigProperty.Builder::build)
-            .collect(Collectors.toMap(LoadedConfigProperty::name, Function.identity()));
+        final Map<String, LoadedConfigProperty> pathProps =
+            response.getData().entrySet().stream()
+                .map(LoadedConfigProperty::withNameAndValue)
+                .map(LoadedConfigProperty.Builder::build)
+                .collect(Collectors.toMap(LoadedConfigProperty::name, Function.identity()));
         result.putAll(pathProps);
       } catch (Exception ex) {
-        LOGGER.warn("unable to load config properties from {}",path, ex);
+        LOGGER.warn("Unable to load config properties from {}", path, ex);
         throw new ConfigSourceNotAvailableException(ex);
       }
     }
     return result;
   }
 
-  /**
-   * This builder method is used internally for test purposes. please use it only for tests. Please
-   * note the following required environment variables are required.
-   *
-   * <ul>
-   *   <li><code>VAULT_SECRETS_PATH</code> is the path to use (defaults to <code>secret</code>)
-   *   <li><code>VAULT_TOKEN</code> is the {@link VaultConfig#token(String) token} to use
-   *   <li><code>VAULT_ADDR</code> is the {@link VaultConfig#address(String) address} of the vault
-   *       (API)
-   * </ul>
-   */
   public static Builder builder() {
     return new Builder();
   }
 
-  /**
-   * This builder method is used internally for test purposes. please use it only for tests
-   *
-   * @param environmentLoader an {@link EnvironmentLoader}
-   */
-  static Builder builder(EnvironmentLoader environmentLoader) {
-    final Builder builder = new Builder();
-    if (environmentLoader != null) {
-      builder.environmentLoader = environmentLoader;
-    }
-    return builder;
-  }
-
   public static final class Builder {
 
-    private Function<VaultInvoker.Builder, VaultInvoker.Builder> vault = Function.identity();
+    private Function<VaultInvoker.Builder, VaultInvoker.Builder> builderFunction = b -> b;
+
     private VaultInvoker invoker;
-    private EnvironmentLoader environmentLoader = VaultInvoker.Builder.ENVIRONMENT_LOADER;
-    private List<String> secretsPaths = new ArrayList<>();
+
+    private List<String> secretsPaths =
+        Optional.ofNullable(ENVIRONMENT_LOADER.loadVariable("VAULT_SECRETS_PATH"))
+            .or(() -> Optional.ofNullable(ENVIRONMENT_LOADER.loadVariable("VAULT_SECRETS_PATHS")))
+            .map(s -> s.split(":"))
```

Comment:
Is it new street magic? Let's add javadoc or use more explicit separator like `,`.